platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dependencies/DefaultProjectDependency.java

```diff
@@ -87,7 +88,7 @@ public Path getIdentityPath() {
     private Configuration findProjectConfiguration() {
         ConfigurationContainer dependencyConfigurations = getDependencyProject().getConfigurations();
         String declaredConfiguration = getTargetConfiguration();
-        Configuration selectedConfiguration = dependencyConfigurations.getByName(GUtil.elvis(declaredConfiguration, Dependency.DEFAULT_CONFIGURATION));
+        Configuration selectedConfiguration = dependencyConfigurations.getByName(java.util.Objects.requireNonNull(GUtil.elvis(declaredConfiguration, Dependency.DEFAULT_CONFIGURATION)));
```

Comment:
I don't think this `requireNonNull` is necessary. Its just the IDE not knowing properly how `GUtil.elvis` works. A proper solution would be to add a [@Contract](https://www.jetbrains.com/help/resharper/Contract_Annotations.html) annotation, explaining to IDEA how this method works nullability wise.

However, I don't think adding `requireNonNull` to shut up the ide is helping here. We might as well use a ternary